From dda5219274bf73f36406599978a6cca36e1b043e Mon Sep 17 00:00:00 2001
Message-ID: <dda5219274bf73f36406599978a6cca36e1b043e.1719691458.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1719691458.git.sharlatanus@gmail.com>
References: <cover.1719691458.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 29 Apr 2024 21:33:43 +0100
Subject: [PATCH 06/26] gnu: Add go-github-com-dsnet-golib.

* gnu/packages/golang-xyz.scm (go-github-com-dsnet-golib): New variable.

Change-Id: If3f4bc017119c05ff837414bf5a839c39feff69b
---
 gnu/packages/golang-xyz.scm | 40 +++++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 7b92ef052f..7e7fcc5105 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -1356,6 +1356,46 @@ (define-public go-github-com-docopt-docopt-go
 implementation of http://docopt.org/.")
       (license license:expat))))
 
+(define-public go-github-com-dsnet-golib
+  (package
+    (name "go-github-com-dsnet-golib")
+    (version "1.0.2")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/dsnet/golib")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1f314wzr16w6ix3bs7ginjkizgyl3b1r3j2gvvqzr8dv53r4s5cq"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/dsnet/golib"))
+    (home-page "https://github.com/dsnet/golib")
+    (synopsis "Collection of helper libraries for Golang")
+    (description
+     "@code{golib} is a collection of unrelated libraries.
+This package provides a following list of Golang models:
+@table @code
+@item bufpipe
+Implements a buffered pipe.
+@item cron
+Parses and runs cron schedules.
+@item hashmerge
+Merges hash checksums.
+@item jsoncs
+Implements JSON Canonicalization Scheme (JCS) as specified in RFC 8785.
+@item jsonfmt
+Implements a JSON formatter.
+@item memfile
+Implements an in-memory emulation of @code{os.File}.
+@item unitconv
+Implements string conversion functionality for unit prefixes.
+@end table")
+    (license license:bsd-3)))
+
 (define-public go-github-com-dustin-gojson
   (package
     (name "go-github-com-dustin-gojson")
-- 
2.41.0

